server/src/controllers/campaign.js

```diff
@@ -80,6 +80,63 @@ class CampaignController {
 
     res.status(200).send();
   };
+
+  static addStep = async (req, res) => {
+    const campaign = await db.Campaign.findOne({
+      where: {
+        id: req.params.id,
+        userId: req.currentUser.id
+      },
+      include: db.Step
+    });
+
+    if (!campaign) {
+      throw new BadRequestError("Campaign does not exist");
+    }
+
+    let nextOrder = 1;
+    if (campaign.Steps.length > 0) {
+      nextOrder = campaign.Steps.length + 1;
+    }
+
+    const step = await db.Step.create({
+      order: nextOrder,
+      campaignId: campaign.id,
+      subject: req.body.subject,
+      body: req.body.body
+    });
+
+    res.status(201).send(step);
+  };
+
+  static updateStep = async (req, res) => {
+    const campaign = await db.Campaign.findOne({
+      where: {
+        id: req.params.id,
+        userId: req.currentUser.id
+      },
+      include: {
+        model: db.Step,
+        required: false,
+        where: {
+          order: req.params.order
+        }
+      }
+    });
```

Comment:
This should be a simple get by order and compaign id - it's more efficient database call then this where statement

Reply:
Updated to get the step using campaign id and order